File: files/300-399/359/CH-359.R
library(tidyverse)
library(readxl)

path <- "300-399/359/CH-359 Replacement.xlsx"
input <- read_excel(path, range = "B3:E10")
test <- read_excel(path, range = "G3:J10")

result = input %>%
  add_count(`Customer ID`) %>%
  mutate(`Customer ID` = ifelse(n == 1, "Other", `Customer ID`)) %>%
  select(-n)

all.equal(result, test)
# [1] TRUE

File: files/300-399/359/CH-359.py
import pandas as pd

path = "300-399/359/CH-359 Replacement.xlsx"
input = pd.read_excel(path, usecols="B:E", skiprows=2, nrows=8)
test = pd.read_excel(path, usecols="G:J", skiprows=2, nrows=8).rename(columns=lambda col: col.replace(".1", ""))

input["Customer ID"] = input["Customer ID"].where(input["Customer ID"].map(input["Customer ID"].value_counts()) != 1, "Other")

print(input.equals(test))
# True